src/components/Search/SelectedRevisionItem.tsx

```diff
@@ -138,7 +151,9 @@ function SelectedRevisionItem({
           name='close-button'
           aria-label='close-button'
           className={`${iconClassName} revision-action close-button`}
-          onClick={handleClose}
+          onClick={
+            isEditable ? handleRemoveEditViewRevision : handleRemoveRevision
```

Comment:
Like suggested above, I would recommend to do this distinction in the parent instead of here, so that we don't have to pass `setInProgress`, and just have a callback `onRemove` on the item.